src/ImageFiltering.jl

```diff
@@ -50,10 +50,16 @@ function Base.transpose(A::StaticOffsetArray{T,2}) where T
 end
 
 module Algorithm
+    import FFTW
     # deliberately don't export these, but it's expected that they
     # will be used as Algorithm.FFT(), etc.
     abstract type Alg end
-    "Filter using the Fast Fourier Transform" struct FFT <: Alg end
+    "Filter using the Fast Fourier Transform" struct FFT <: Alg
+        plan1::Union{FFTW.rFFTWPlan,Nothing}
```

Comment:
Probably `rFFTWPlan` is not a concrete type?  So should this be a parametric struct?
(I might be using the terms incorrectly...)

Reply:
This PR isn't in a working state yet. My plan is to bring the in-place functions over from RFFT.jl

Reply:
If one can move RFFT from one registry to another, I'm happy to do that if it would help.

Reply:
Thanks but I was going to suggest that once/if I got it working through just copying the code into here first